tests/run-make/rustdoc-map-file/rmake.rs

```diff
@@ -8,6 +10,48 @@ fn main() {
         .arg("--generate-redirect-map")
         .out_dir(&out_dir)
         .run();
-    // FIXME (GuillaumeGomez): Port the python script to Rust as well.
-    python_command().arg("validate_json.py").arg(&out_dir).run();
+
+    let mut found_file = false;
+    read_dir_entries_recursive(&out_dir, |path| {
+        if !found_file
+            && path.is_file()
+            && path.file_name().map(|name| name == "redirect-map.json").unwrap_or(false)
+        {
+            found_file = true;
+            let generated = read_to_string(path);
+            let expected = read_to_string("expected.json");
+            let generated = jzon::parse(&generated).expect("failed to parse JSON");
+            let expected = jzon::parse(&expected).expect("failed to parse JSON");
+
+            let mut differences = Vec::new();
+            for (key, expected_value) in expected.entries() {
+                match generated.get(key) {
+                    Some(value) => {
+                        if expected_value != value {
+                            differences.push(format!("values for key `{key}` don't match"));
```

Comment:
Question: do we also want to provide info on *how* they don't match upon test failure?

Reply:
Good idea!